EXRPdfReader/Views/PdfDataSource.m

```diff
@@ -0,0 +1,77 @@
+//
+//  PdfDataSource.m
+//  EXRPdfReader
+//
+//  Created by Arnold Joseph Caesar Esteban on 1/13/21.
+//  Copyright © 2021 Arnold Joseph Caesar Esteban. All rights reserved.
+//
+
+#import "PdfDataSource.h"
+#import "PdfContentViewController.h"
+
+@interface PdfDataSource ()
+
+@property (assign, nonatomic) CGPDFDocumentRef pdf;
+
+@property (assign, nonatomic) NSInteger numberOfPages;
+
+@end
+
+@implementation PdfDataSource
+
+- (instancetype)initWithFilePath:(NSString *)filePath {
+    if (self = [super init]) {
+        NSURL *pdfURL = [[NSBundle mainBundle] URLForResource:filePath
+                                                withExtension:nil];
+        if (pdfURL) {
+            NSLog(@"PdfDataSource: Successfully loaded PDF file");
+            self.pdf = CGPDFDocumentCreateWithURL((__bridge CFURLRef) pdfURL);
+            self.numberOfPages = CGPDFDocumentGetNumberOfPages(self.pdf);
+        } else {
+            NSLog(@"PdfDataSource: Missing pdf file %@", filePath);
+        }
+    }
+    return self;
+}
+
+#pragma mark - Navigation Methods
+
+- (PdfContentViewController *)viewControllerAtIndex:(NSUInteger)index {
+    // Create a new view controller and pass suitable data.
```

Comment:
Best to add this as a method comment so that other devs will know that this method actually creates a new VC